src/adapter/src/command.rs

```diff
@@ -803,14 +830,20 @@ impl ExecuteResponse {
             DropOwned => vec![DroppedOwned],
             PlanKind::EmptyQuery => vec![ExecuteResponseKind::EmptyQuery],
             Explain | Select | ShowAllVariables | ShowCreate | ShowVariable | InspectShard => {
-                vec![CopyTo, SendingRows]
+                vec![CopyTo, SendingRows, SendingRowsImmediate]
             }
-            Execute | ReadThenWrite => vec![Deleted, Inserted, SendingRows, Updated],
+            Execute | ReadThenWrite => vec![
+                Deleted,
+                Inserted,
+                SendingRows,
+                SendingRowsImmediate,
+                Updated,
+            ],
             PlanKind::Fetch => vec![ExecuteResponseKind::Fetch],
             GrantPrivileges => vec![GrantedPrivilege],
             GrantRole => vec![GrantedRole],
             CopyRows => vec![Inserted],
-            Insert => vec![Inserted, SendingRows],
+            Insert => vec![Inserted, SendingRows, SendingRowsImmediate],
```

Comment:
```suggestion
            Insert => vec![Inserted, SendingRowsImmediate],
```

Pretty sure this will always be the case? If I'm wrong tests will panic.